conformance/tests/httproute-request-multiple-mirrors.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteRequestMirrors)
+}
+
+var HTTPRouteRequestMirrors = suite.ConformanceTest{
+	ShortName:   "HTTPRouteRequestMirrors",
+	Description: "An HTTPRoute with multiple request mirror filters",
+	Manifests:   []string{"tests/httproute-request-multiple-mirrors.yaml"},
+	Features: []suite.SupportedFeature{
+		suite.SupportGateway,
+		suite.SupportHTTPRoute,
+		suite.SupportHTTPRouteRequestMirror,
+		suite.SupportHTTPRouteRequestMirrors,
+	},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		ns := "gateway-conformance-infra"
+		routeNN := types.NamespacedName{Name: "request-multiple-mirrors", Namespace: ns}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+
+		testCases := []http.ExpectedResponse{
+			{
+				Request: http.Request{
+					Path: "/mirror",
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/mirror",
+					},
+				},
+				Backend: "infra-backend-v1",
+				MirroredTo: []http.BackendRef{
+					{
+						Name:      "infra-backend-v2",
+						Namespace: ns,
+					},
+					{
+						Name:      "infra-backend-v3",
+						Namespace: "another-namespace",
+					},
+				},
+				Namespace: ns,
```

Comment:
Not a requirement for this PR, but I am curious if changing `Backend` to the new `BackendRef` struct you created is something we want and how much effort it will take (since it is widely used in a lot of tests).

Reply:
Well, if there is a demand for it I can address it in a follow-up PR

Reply:
would be good if you can do a quick check and see if its something we would want and if so, open an issue for it (it can be a good-first-issue imo)